sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/AppConfigurationReplicaClient.java

```diff
@@ -134,6 +141,38 @@ List<ConfigurationSetting> listSettings(SettingSelector settingSelector)
         }
     }
 
+    FeatureFlags listFeatureFlags(SettingSelector settingSelector)
+        throws HttpResponseException {
+        List<ConfigurationSetting> configurationSettings = new ArrayList<>();
+        List<MatchConditions> checks = new ArrayList<>();
+        try {
+            client.listConfigurationSettings(settingSelector).streamByPage().forEach(pagedResponse -> {
+                checks.add(
+                    new MatchConditions().setIfNoneMatch(pagedResponse.getHeaders().getValue(HttpHeaderName.ETAG)));
```

Comment:
what is this `checks` used for? To check if there's a "ETAG" header?

Reply:
`checks` is a list of `MatchConditions`. The `MatchConditions` class contains the logic from the SDK for paged tags. if you look down a few lines we add the `checks` list to the `SettingSelector`. This modified `SettingSelector` will be the one used in refresh to check for changes.